odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,169 @@ def __repr__(self):
                                          self.__norm.pwisenorm.exponent)
 
 
+class IndicatorSimplex(Functional):
+
+    """The indicator functional of a simplex in any dimension.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \\begin{cases}
+            0 & \\text{if } x_i \geq 0 \\forall i \\text{ and } \sum_i x_i = r
+            \\\\
+            +\\infty & \\text{else.}
+        \\end{cases}
+
+    where `r` is the radius.
+    """
+
+    def __init__(self, space, radius=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        radius : positive float, optional
+            Radius `r` of simplex.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+        self.radius = radius
+
+    def __call__(self, x):
+        """Return whether x is in the unit simplex."""
+
+        r = self.radius
+        if (x.ufuncs.sum() == r and x.ufuncs.greater_equal(0).asarray().all()):
+            return 0
+
+        return np.inf
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional. The indicator functional is not
```

Comment:
One line (or split)

Reply:
What do you mean exactly?

Reply:
The summary line should be one line. If more explanation is necessary, that should be in a new block of text, separated by a blank line.
We're largely following [this convention](https://www.python.org/dev/peps/pep-0257/).

Of course, one can argue about the usefulness of summary lines like "Gradient operator of the functional." But I have yet to see better solutions.